### ISSUE TITLE
Re-write React Challenge 30: Create a Controlled Form

### ISSUE THREAD

**Issue by bonham000** [bonham000/fcc-react-tests-module#225]
*Copied over from [original issue in FCC repo](https://github.com/freeCodeCamp/freeCodeCamp/issues/15690).*

#### Challenge Name

React 30: Create a Controlled Form

#### Issue Description
The instructions for the challenge is as follows:

> The MyForm component is set up to handle changes on an input element. Add the input element in the return of the MyForm component, setting its value and onChange() attributes like the last challenge. Then, create a button with the text "Submit" to submit the input value. (The form won't submit anywhere, this is a simulation). The button should have an onClick() handler which triggers a method called handleSubmit(). This method should take the value that's currently in the input and set it to the submit property in local state. 

> Finally, create an h1 tag below the button which renders the submit value from the component's state. You can type in the form and click the button, and you should see your input rendered to the page.

I believe this challenge will teach students bad practice. In this form, the input element does not respond to the user pressing 'Enter' to submit the form.

I propose the solution require students to use a `<form>` tag with an `onSubmit` prop that is set to the `handleSubmit` function.  This is what is used in the [official React documentation](https://facebook.github.io/react/docs/forms.html#controlled-components).

#### Current Challenge Solution

```jsx
class MyForm extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			input: '',
			submit: ''
		};
	}
	handleChange = (event) => {
		this.setState({
			input: event.target.value
		});
	}
	handleSubmit = () => {
		this.setState({
			submit: this.state.input
		});
	}
	render() {
  	  return (
	    <div>
               <input
        	  value={this.state.input}
        	  onChange={this.handleChange} />
                 <button onClick={this.handleSubmit}>Submit!</button>
                 <h1>{this.state.submit}</h1>
	    </div>
          );
      }
};
```
#### Proposed Challenge Solution
```jsx
class MyForm extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			input: '',
			submit: ''
		};
	}
	handleChange = (event) => {
		this.setState({
			input: event.target.value
		});
	}
	handleSubmit = () => {
		this.setState({
			submit: this.state.input
		});
	}
	render() {
  	  return (
	    <form onSubmit={this.handleSubmit}>
              <input
        	value={this.state.input}
        	onChange={this.handleChange} />
              <button>Submit!</button>
              <h1>{this.state.submit}</h1>
	    </form>
         );
      }
};
```

**Comment by AryanJ-NYC** [comment]
I attempted to fix this but #227 makes this difficult.

**Comment by bonham000** [comment]
I'll close this and let #227 track it.
